tests/template/test_load_context.py

```diff
@@ -51,14 +51,42 @@ def test_valid_context(self, fake_repo_path):
         assert str(fake_repo_path.resolve() / "src") in sys.path
 
     def test_valid_context_with_env(self, mocker, monkeypatch, fake_repo_path):
-        """Test getting project context when Kedro config environment is
-        specified in the environment variable.
-        """
+        """Test getting project context when Kedro config environment is specified
+        in the environment variable."""
+        # Disable logging.config.dictConfig in KedroContext._setup_logging as
+        # it changes logging.config and affects other unit tests
+        mocker.patch("logging.config.dictConfig")
         mocker.patch("kedro.config.config.ConfigLoader.get")
         monkeypatch.setenv("KEDRO_ENV", "my_fake_env")
         result = load_context(str(fake_repo_path))
         assert result.env == "my_fake_env"
 
+    def test_loading_valid_context_twice(self, mocker, fake_repo_path):
+        """Test getting project context twice."""
+        mocker.patch("logging.config.dictConfig")
+
+        # first time
+        result_first_time = load_context(str(fake_repo_path))
+        assert result_first_time.project_name == "Test Project"
+        assert result_first_time.project_version == kedro.__version__
+        assert str(fake_repo_path.resolve() / "src") in sys.path
+
+        # second time
+        try:
+            result_second_time = load_context(str(fake_repo_path))
+            assert result_first_time.project_path == result_second_time.project_path
+            assert (
+                result_first_time.project_version == result_second_time.project_version
+            )
+            assert str(fake_repo_path.resolve() / "src") in sys.path
+        except KedroContextError as kce:
+            print(
+                "FAILED: Should have loaded the context twice successfully, "
+                "and not got this exception:",
+                kce,
+            )
+            assert 0
```

Comment:
What is this doing? Can't you just load the context twice? If an exception is raised, the test will fail. If you want to provide more information, I'd replace the `print` and `assert 0` with `pytest.fail(your_failure_reason)`

Reply:
Good point I can make it condensed.

Reply:
Applied via 9b9897